src/main/java/at/hannibal2/skyhanni/utils/renderables/ScrollInput.kt

```diff
@@ -22,8 +22,14 @@ abstract class ScrollInput(
         }
         get() = scrollValue.getValue()
 
+    enum class ScrollDirection(val value: Int) {
+        UP(-1),
+        DOWN(1),
+    }
+
     fun asInt() = scroll.toInt()
     fun asDouble() = scroll
+    fun asDirection() = ScrollDirection.entries.firstOrNull { it.value == scroll.toInt() }
```

Comment:
This is not really clean. A x >0 / x < 0 checks makes more sense. Since this one is callable by all scroll inputs and only makes sense for the pure variant the way you implemented it.
So make it a three state enum and use:
```suggestion
    fun asDirection() = if(scroll == 0.0) NONE else if(scroll > 0.0) DOWN else UP
```
Also btw why is this inverted? Why is -1 up and 1 down?

Reply:
```kotlin
            private fun ScrollInput.Companion.PureVertical.tryUpdateScroll() {
                pureScrollInput.update(true)
                when (asInt()) {
                    -1 -> onClick(Direction.RIGHT) // Scroll Up -> Right Click
                    1 -> onClick(Direction.LEFT) // Scroll Down -> Left Click
                    else -> return
                }
                dispose()
            }
 ```
 
 That's existing code, -1 indicates up, 1 indicates down 🤷